tests/nativeapp/test_run_processor.py

```diff
@@ -131,11 +152,70 @@ def drop_application_before_upgrade(cascade: bool = False):
     )
 
 
+test_pdf = dedent(
+    """\
+        definition_version: 2
+        entities:
+            app_pkg:
+                type: application package
+                stage: app_src.stage
+                manifest: app/manifest.yml
+                artifacts:
+                    - setup.sql
+                    - src: app/manifest.yml
+                      dest: manifest.yml
+                meta:
+                    role: package_role
+                    warehouse: pkg_warehouse
+            myapp:
+                type: application
+                debug: true
+                from:
+                    target: app_pkg
+                meta:
+                    role: app_role
+                    warehouse: app_warehouse
+    """
+)
+
+
+def setup_project_file(current_working_directory: str, pdf=None):
```

Comment:
let's use project factories in here

Reply:
I would like to keep this for a separate PR if possible.